mesonbuild/modules/external_project.py

```diff
@@ -201,6 +201,9 @@ def _format_options(self, options: T.List[str], variables: T.List[T.Tuple[str, s
     def _run(self, step: str, command: T.List[str], workdir: Path) -> None:
         mlog.log(f'External project {self.name}:', mlog.bold(step))
         m = 'Running command ' + str(command) + ' in directory ' + str(workdir) + '\n'
+        m += 'With environment variables:\n'
+        for k, v in sorted(self.run_env.items()):
+            m += f'  {k}={v}\n'
```

Comment:
With `verbose: true` in particular, this is excessively long since it contains the full contents of `os.environ`

Reply:
Would you be fine with printing all the env if verbose is false?

Reply:
If it's just going into the logfile that's fine, yeah. And, it could even be helpful, perhaps. We also add the entire "Inherited environment:" in testlog.txt for the same reason, it potentially helps you analyze issues retroactively, in a new shell session with different environment variables.

mlog.log also copies itself to `builddir/meson-logs/meson-log.txt`. You can use `mlog.debug` to print *only* to that logfile since `{project}-configure.log` doesn't get created with `verbose: true`.

Reply:
Okay, I've now changed it print the env only to the log file. I didn't use `mlog.debug` since my understanding is that it prints into the main meson log file, while I think its better to have it only in the `project-configure/build/install.log` file that is specific for external projects.